src/Mod/Path/PathTests/TestPathSlot.py

```diff
@@ -0,0 +1,264 @@
+        """
+        # FreeCAD.Console.PrintMessage("TestPathSlot.tearDownClass()\n")
+        FreeCAD.closeDocument(FreeCAD.ActiveDocument.Name)
+
+    def setUp(self):
+        """setUp()...
+        This method is called prior to each test() method.  Add code and objects here
+        that are needed for multiple test() methods.
+        """
+        self.doc = FreeCAD.ActiveDocument
+        self.con = FreeCAD.Console
+
+    def tearDown(self):
+        """tearDown()...
+        This method is called after each test() method. Add cleanup instructions here.
+        Such cleanup instructions will likely undo those in the setUp() method.
+        """
+        pass
+
+    def _set_depths_and_heights(self, op):
+        # Set Depths
+        # set start and final depth in order to eliminate effects of stock (and its default values)
+        op.setExpression('StartDepth', None)
+        op.StartDepth.Value = 10.0
+        op.setExpression('FinalDepth', None)
+        op.FinalDepth.Value = 5.0
+        op.setExpression('StepDown', None)
+        op.StepDown.Value = 5.0
+
+        # Set Heights
+        # default values used
+
+    def _format_point(self, cmd):
+        """Accepts command dict and returns point string coordinate"""
+        x = round(cmd["X"], 2)
+        y = round(cmd["Y"], 2)
+        return "({}, {})".format(x, y)
+
+    def test00(self):
+        '''Test Slot on horizontal rectangular face: ReverseDirection=True; LayerMode=Single-pass'''
+
+        # Identify base feature(s) to be used for operation's Base Geometry
+        base = self.doc.Cut
+        for i in range(11):
+            face = "Face%d" % (i+1)
+            f = base.Shape.getElement(face)
+            if (f.Surface.Axis == FreeCAD.Vector(0,0,1) and
+                f.Orientation == 'Forward' and f.BoundBox.ZMin == 5.0):
+                break
+
+        # Instantiate a Slot operation and sets standard depths & heights
+        slot = PathSlot.Create('Slot-test00')
+        self._set_depths_and_heights(slot)
+
+        # Set Base Geometry
+        slot.Base = (base, [face])
+
+        # Set Operation
+        slot.ReverseDirection = True
+        slot.LayerMode = "Single-pass"
+
+        slot.recompute()
+        self.doc.recompute()
+        
+        pnts = list()
+        for c in slot.Path.Commands:
+            p = c.Parameters
+            keys = str(p.keys())[9:]
+            if "'X'" in keys and "'Y'" in keys:
+                pnts.append(self._format_point(p))
+        pnts.sort()
+        # print("pnts: {}\n".format(pnts))
+
+        # Verify point count
+        if len(pnts) != 2:
+            for c in slot.Path.Commands:
+                p = c.Parameters
+                print("Parameters: {}".format(p))
+            self.assertEqual(len(pnts), 2)
+
+        # Verify each line-segment point, excluding arcs
+        verify_points = ['(7.5, 0.0)', '(7.5, 15.0)']
+        for i in range(2):
+            self.assertEqual(pnts[i], verify_points[i])
```

Comment:
I highly recommend not comparing doubles directly but use PathGeom to make the test more robust.

Reply:
In this instance, the `self.assertEqual()` is comparing strings as I have seen done elsewhere in Path unit tests.  Should I compare strings differently?

Reply:
you're converting a float to a string - could you end up with something like `'(7.499999999999, 0.00000000001)'` ?

Reply:
> you're converting a float to a string - could you end up with something like `'(7.499999999999, 0.00000000001)'` ?

I don't think so.  I am rounding values to say three or six decimal places, prior to comparison, to avoid such scenarios.  So long as the `round()` function is working as I am expecting, the subsequent string representation should be only, say, two decimal places max per my rounding argument used.

Does this response address your concern with my methodology, or am I still missing something in the process?